pkg/config/install_requirements.go

```diff
@@ -293,16 +293,39 @@ type ClusterConfig struct {
 	HelmMajorVersion string `json:"helmMajorVersion,omitempty"`
 }
 
+// VaultConfig contains Vault configuration for boot
 type VaultConfig struct {
-	// Name the name of the vault if using vault for secretts
+	// Name the name of the vault if using vault for secrets
 	Name           string `json:"name,omitempty"`
 	Bucket         string `json:"bucket,omitempty"`
 	Keyring        string `json:"keyring,omitempty"`
 	Key            string `json:"key,omitempty"`
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 	RecreateBucket bool   `json:"recreateBucket,omitempty"`
+	AWSIAMUserARN  string `json:"awsIAMUserARN,omitempty"`
 	// Optionally allow us to override the default lookup of the Vault URL, could be incluster service or external ingress
-	DisableURLDiscovery bool `json:"disableURLDiscovery,omitempty"`
+	DisableURLDiscovery bool           `json:"disableURLDiscovery,omitempty"`
+	AWSConfig           VaultAWSConfig `json:",omitempty"`
```

Comment:
This config seems to be inconsistent and confusing. We have `Bucket` in the top structure but also `AWSConfig.S3Bucket` in a substructure. Also the `AWSIAMUserARN` should be rather moved into `AWSConfig` section.

We should move the GCP related config into something like GCPConfig, otherwise the users will be confused.

Reply:
You are right, AWSIAMUserARN should be removed because it's already in AWSConfig, I forgot to remove it.

Shouldn't the change of GCP configuration be done in a different PR? The problem now when working with some provider is that you also have to modify a bunch of things for the other provider and it takes ages to close something.

Reply:
We should streamline the configuration for each provider since we are still in the early stage. Otherwise we introduce more stuff which it will be quite difficult to deprecate later.